includes/Modules/Analytics/Advanced_Tracking/Measurement_Events/Woocommerce_Event_List.php

```diff
@@ -0,0 +1,282 @@
+				'selector'   => 'div.wc-proceed-to-checkout .checkout-button',
+				'on'         => 'click',
+				'metadata'   => <<<CALLBACK
+function( params, element ) {
+	var currency = woocommerceStoreData.currency;
+	params['currency'] = currency;
+
+	var value = 0.0;
+	items = [];
+	cartItems = document.querySelectorAll( '.woocommerce-cart-form__cart-item' );
+	for ( cartItem of cartItems ) {
+		var productName = cartItem.querySelector( '.product-name a' ).innerText;
+
+		var item = woocommerceProducts[ productName ];
+		item['quantity'] = woocommerceCartData.item_quantities[ productName ];
+		items.push( item );
+
+		var productQuantity = woocommerceCartData.item_quantities[ productName ];
+		value += productQuantity * parseFloat( item['price'] );
+	}
+	params['value'] = value;
+	params['items'] = items;
+
+	return params;
+}
+CALLBACK
+			,
+			)
+		);
+		$this->add_event( $event );
+
+		$event = new Measurement_Event(
+			array(
+				'pluginName' => 'Woocommerce',
+				'category'   => 'ecommerce',
+				'action'     => 'view_cart',
+				'selector'   => 'a.added_to_cart.wc-forward',
+				'on'         => 'click',
+			)
+		);
+		$this->add_event( $event );
+
+		$event = new Measurement_Event(
+			array(
+				'pluginName' => 'Woocommerce',
+				'category'   => 'ecommerce',
+				'action'     => 'view_cart',
+				'selector'   => 'div.woocommerce-message a.wc-forward',
+				'on'         => 'click',
+			)
+		);
+		$this->add_event( $event );
+
+		$event = new Measurement_Event(
+			array(
+				'pluginName' => 'Woocommerce',
+				'category'   => 'ecommerce',
+				'action'     => 'view_cart',
+				'selector'   => 'a.cart-contents',
+				'on'         => 'click',
+			)
+		);
+		$this->add_event( $event );
+
+		$event = new Measurement_Event(
+			array(
+				'pluginName' => 'Woocommerce',
+				'category'   => 'ecommerce',
+				'action'     => 'update_cart',
+				'selector'   => '.woocommerce-cart-form__contents .coupon ~ .button',
+				'on'         => 'click',
+				'metadata'   => <<<CALLBACK
+function( params, element ) {
+	cartItems = document.querySelectorAll( '.woocommerce-cart-form__cart-item' );
+	for ( cartItem of cartItems ) {
+		var productName = cartItem.querySelector( '.product-name a' ).innerText;
+		var newQuantity = cartItem.querySelector( '.product-quantity input' ).valueAsNumber;
+		woocommerceCartData.item_quantities[ productName ] = newQuantity;
+	}
+	return params;
+}
+CALLBACK
+			,
+			)
+		);
+		$this->add_event( $event );
+
+		$event = new Measurement_Event(
+			array(
+				'pluginName' => 'Woocommerce',
+				'category'   => 'ecommerce',
+				'action'     => 'view_item',
+				'selector'   => '.content-area a.woocommerce-LoopProduct-link',
+				'on'         => 'click',
+			)
+		);
+		$this->add_event( $event );
+
+		$event = new Measurement_Event(
+			array(
+				'pluginName' => 'Woocommerce',
+				'category'   => 'ecommerce',
+				'action'     => 'purchase',
+				'selector'   => '', /*'.woocommerce-page form.woocommerce-checkout'*/
```

Comment:
Let's for now put the purchase button selector here and add a `TODO` to iterate on that?